pkg/api/util/names.go

```diff
@@ -37,7 +39,13 @@ func ComputeCertificateRequestName(crt *cmapi.Certificate) (string, error) {
 		return "", err
 	}
 
-	// shorten the cert name to 52 chars to ensure the total length of the name
-	// is less than or equal to 64 characters
-	return fmt.Sprintf("%.52s-%d", crt.Name, hashF.Sum32()), nil
+	if len(crt.Name) >= 52 {
+		// shorten the cert name to 52 chars to ensure the total length of the name
+		// also shorten the 52 char string to the last non-symbol character
+		// is less than or equal to 64 characters
+		validCharIndexes := regexp.MustCompile(`[a-zA-Z\d]`).FindAllStringIndex(fmt.Sprintf("%.52s", crt.Name), -1)
+		crt.Name = crt.Name[:validCharIndexes[len(validCharIndexes)-1][1]]
```

Comment:
We should actually modify the passed in Certificate here - instead, just trim the name in the *returned* string

Reply:
Changed it in 0e1f64a33746bd3f839c42ab6daff5f16a004232
Also removed the unused `crt.DeepCopy()`